diff -up ksh-20120801/src/cmd/ksh93/sh/xec.c.hokaido ksh-20120801/src/cmd/ksh93/sh/xec.c
--- ksh-20120801/src/cmd/ksh93/sh/xec.c.hokaido	2014-06-10 08:30:29.953485752 +0200
+++ ksh-20120801/src/cmd/ksh93/sh/xec.c	2014-06-10 09:22:32.646799331 +0200
@@ -1618,11 +1618,25 @@ int sh_exec(register const Shnode_t *t,
 #endif /* SHOPT_COSHELL */
 			if(shp->subshell)
 			{
-				sh_subtmpfile(shp);
+				if(shp->comsub==1)
+				{
+					int t=shp->comsub; 
+					if(t==1)shp->comsub=2;
+					sh_subtmpfile(shp);
+					shp->comsub=t;
+				}
+				else
+					sh_subtmpfile(shp);
 				if(shp->comsub==1 && !(shp->fdstatus[1]&IONOSEEK))
-					unpipe=iousepipe(shp);
+					unpipe = iousepipe(shp);
 				if((type&(FAMP|TFORK))==(FAMP|TFORK))
-					sh_subfork();
+				{
+					if(shp->comsub && !(shp->fdstatus[1]&IONOSEEK))
+					{
+						if (!unpipe) unpipe = iousepipe(shp);
+						sh_subfork();
+					}
+				}
 			}
 			no_fork = !ntflag && !(type&(FAMP|FPOU)) && !shp->subshell &&
 			    !(shp->st.trapcom[SIGINT] && *shp->st.trapcom[SIGINT]) &&
