--- ksh-20070328/src/cmd/ksh93/sh/xec.c.orig	2007-04-19 08:54:10.000000000 +0200
+++ ksh-20070328/src/cmd/ksh93/sh/xec.c	2007-04-19 08:55:56.000000000 +0200
@@ -1190,6 +1190,8 @@
 			}
 			sh_popcontext(&buff);
 			sh_iorestore(buff.topfd,jmpval);
+			if(buff.olist)
+				free_list(buff.olist);
 			if(type&FPIN)
 			{
 				job.waitall = waitall;
@@ -2878,6 +2880,8 @@
 	else
 		exitset();
 	sh_popcontext(&buff);
+	if(buff.olist)
+		free_list(buff.olist);
 #ifdef SIGTSTP
 	if(jobwasset)
 	{
