--- ksh-20070328/src/lib/libcmd/uname.c.uname_fix	2007-04-17 13:39:44.000000000 +0200
+++ ksh-20070328/src/lib/libcmd/uname.c	2007-04-17 13:41:24.000000000 +0200
@@ -446,11 +446,7 @@
 		if (flags & OPT_domain)
 		{
 			if (!*(s = astconf("SRPC_DOMAIN", NiL, NiL)))
-#if _lib_getdomainname
-				getdomainname(s, sizeof(buf));
-#else
-				/*NOP*/;
-#endif
+				s = ut.__domainname;
 			output(OPT_domain, s, "domain");
 		}
 #if _mem_m_type_utsname
