diff -up ksh-20100309/src/cmd/ksh93/sh/path.c.pathcrash ksh-20100309/src/cmd/ksh93/sh/path.c
--- ksh-20100309/src/cmd/ksh93/sh/path.c.pathcrash	2010-05-05 14:05:41.670681132 +0200
+++ ksh-20100309/src/cmd/ksh93/sh/path.c	2010-05-05 14:05:41.707584330 +0200
@@ -1437,6 +1437,7 @@ static Pathcomp_t *path_addcomp(Pathcomp
 	}
 	for(pp=first, oldpp=0; pp; oldpp=pp, pp=pp->next);
 	pp = newof((Pathcomp_t*)0,Pathcomp_t,1,len+1);
+	pp->shp = sh_getinterp();
 	pp->refcount = 1;
 	memcpy((char*)(pp+1),name,len+1);
 	pp->name = (char*)(pp+1);
