diff -up ksh-20120801/src/cmd/ksh93/sh/xec.c.bigsubst ksh-20120801/src/cmd/ksh93/sh/xec.c
--- ksh-20120801/src/cmd/ksh93/sh/xec.c.bigsubst	2014-02-11 08:54:30.113556940 +0100
+++ ksh-20120801/src/cmd/ksh93/sh/xec.c	2014-02-11 08:54:30.141556816 +0100
@@ -1772,7 +1772,7 @@ int sh_exec(register const Shnode_t *t,
 					}
 					if(shp->pipepid)
 						shp->pipepid = parent;
-					else
+					else if(shp->comsub!=1)
 					{
 						job_wait(parent);
 						if(parent==shp->spid)
