diff -up ksh-20071105/src/cmd/ksh93/sh/init.c.optind ksh-20071105/src/cmd/ksh93/sh/init.c
--- ksh-20071105/src/cmd/ksh93/sh/init.c.optind	2008-01-30 07:56:28.000000000 +0100
+++ ksh-20071105/src/cmd/ksh93/sh/init.c	2008-01-30 07:57:19.000000000 +0100
@@ -198,6 +198,8 @@ static void put_optindex(Namval_t* np,co
 	Shell_t *shp = ((struct shell*)fp)->sh;
 	shp->st.opterror = shp->st.optchar = 0;
 	nv_putv(np, val, flags, fp);
+	if(!val)
+		nv_disc(np,fp,NV_POP);
 }
 
 static Sfdouble_t nget_optindex(register Namval_t* np, Namfun_t *fp)
