diff -up ksh-20100309/src/cmd/ksh93/bltins/whence.c.fixwh ksh-20100309/src/cmd/ksh93/bltins/whence.c
--- ksh-20100309/src/cmd/ksh93/bltins/whence.c.fixwh	2008-09-19 16:13:40.000000000 +0200
+++ ksh-20100309/src/cmd/ksh93/bltins/whence.c	2010-04-29 12:03:22.744013139 +0200
@@ -234,7 +234,10 @@ static int whence(Shell_t *shp,char **ar
 				}
 			}
 			if(flags&Q_FLAG)
+			{
 				r |= !cp;
+				pp = 0;
+			}
 			else if(cp)
 			{
 				if(flags&V_FLAG)
