diff -up ksh-20110208/src/cmd/ksh93/bltins/typeset.c.oddarrays ksh-20110208/src/cmd/ksh93/bltins/typeset.c
--- ksh-20110208/src/cmd/ksh93/bltins/typeset.c.oddarrays	2010-12-31 01:15:57.000000000 +0100
+++ ksh-20110208/src/cmd/ksh93/bltins/typeset.c	2011-03-29 07:21:03.484519769 +0200
@@ -587,7 +587,7 @@ static int     b_common(char **argv,regi
 				path_alias(np,path_absolute(shp,nv_name(np),NIL(Pathcomp_t*)));
 				continue;
 			}
-			np = nv_open(name,troot,nvflags|((nvflags&NV_ASSIGN)?0:NV_ARRAY)|NV_FARRAY);
+			np = nv_open(name,troot,nvflags|((nvflags&NV_ASSIGN)?0:NV_ARRAY)|(iarray?NV_FARRAY:0));
 			if(nv_isnull(np) && !nv_isarray(np) && nv_isattr(np,NV_NOFREE))
 				nv_offattr(np,NV_NOFREE);
 			if(tp->pflag)
