--- ksh-20070328/src/cmd/ksh93/data/builtins.c.builtoff	2007-05-22 13:14:33.000000000 +0200
+++ ksh-20070328/src/cmd/ksh93/data/builtins.c	2007-05-22 13:14:51.000000000 +0200
@@ -128,6 +128,7 @@
 #ifdef SHOPT_CMDLIB_HDR
 #include SHOPT_CMDLIB_HDR
 #else
+#if 0
 	CMDLIST(basename)
 	CMDLIST(chmod)
 	CMDLIST(dirname)
@@ -141,6 +142,7 @@
 	CMDLIST(wc)
 	CMDLIST(sync)
 #endif
+#endif
 	"",		0, 0 
 };
 
